test/test.jl

```diff
@@ -92,73 +95,97 @@ end
+            @test 1 == 1
+            @test 2 == :foo
+            @test 3 == 3
+            @testset "d" begin
+                @test 4 == 4
+            end
+            @testset begin
+                @test :blank != :notblank
+            end
         end
-    end
-    @testset "loops without desc" begin
-        @testset for T in (Float32, Float64)
-            @test 1 == T(1)
+        @testset "inner1" begin
+            @test 1 == 1
+            @test 2 == 2
+            @test 3 == :bar
+            @test 4 == 4
+            @test_throws ErrorException 1+1
+            @test_throws ErrorException error()
+            @test_throws RemoteException error()
+            @testset "errrrr" begin
+                @test "not bool"
+                @test error()
+            end
+
+            error("exceptions in testsets should be caught")
+            @test 1 == 1 # this test will not be run
         end
-        @testset for T in (Float32, Float64), S in (Int32,Int64)
-            @test S(1) == T(1)
+
+        @testset "loop with desc" begin
+            @testset "loop1 $T" for T in (Float32, Float64)
+                @test 1 == T(1)
+            end
         end
-    end
-    srand(123)
-    @testset "some loops fail" begin
-        @testset for i in 1:5
-            @test i <= rand(1:10)
+        @testset "loops without desc" begin
+            @testset for T in (Float32, Float64)
+                @test 1 == T(1)
+            end
+            @testset for T in (Float32, Float64), S in (Int32,Int64)
+                @test S(1) == T(1)
+            end
         end
-        # should add 3 errors and 3 passing tests
-        @testset for i in 1:6
-            iseven(i) || error("error outside of test")
-            @test true # only gets run if the above passed
+        srand(123)
+        @testset "some loops fail" begin
+            @testset for i in 1:5
+                @test i <= rand(1:10)
+            end
+            # should add 3 errors and 3 passing tests
+            @testset for i in 1:6
+                iseven(i) || error("error outside of test")
+                @test true # only gets run if the above passed
+            end
         end
     end
+    catch ex
+        #redirect_stdout(OLD_STDOUT)
+        #redirect_stderr(OLD_STDERR)
+        #@show ex
+    end
+    @testset "ts results" begin
+        @test isa(ts, Test.DefaultTestSet)
+        passes, fails, errors, broken, c_passes, c_fails, c_errors, c_broken = Base.Test.get_test_counts(ts)
+        total_pass   = passes + c_passes
+        total_fail   = fails  + c_fails
+        total_error  = errors + c_errors
+        total_broken = broken + c_broken
+        @test total_pass   == 24
+        @test total_fail   == 6
+        @test total_error  == 6
+        @test total_broken == 0
+    end
+    ts.anynonpass = false
+    deleteat!(Base.Test.get_testset().results,1)
 end
-# These lines shouldn't be called
-redirect_stdout(OLD_STDOUT)
-error("No exception was thrown!")
```

Comment:
but the "These lines shouldn't be called" and this error back right before the catch